yt_dlp/extractor/arte.py

```diff
@@ -181,13 +195,18 @@ def _real_extract(self, url):
                 self.report_warning(f'Skipping stream with unknown protocol {stream["protocol"]}')
 
             # TODO: chapters from stream['segments']?
-            # The JS also looks for chapters in config['data']['attributes']['chapters'],
-            # but I am yet to find a video having those
 
         formats.extend(secondary_formats)
         self._remove_duplicate_formats(formats)
 
         metadata = config['data']['attributes']['metadata']
+        chapters = traverse_obj(config, ('data', 'attributes', 'chapters', 'elements'), expected_type=list)
+        if chapters:
+            chapters = [{
+                'start_time': chap.get('startTime'),
+                'end_time': next_chap.get('startTime'),
+                'title': chap.get('title')
+            } for chap, next_chap in zip(chapters, chapters[1:])]
```

Comment:
this code omits the last chapter in the API response.

the core code will set the `end_time` for you, so you can do this instead:

```suggestion
        chapters = []
        for chapter in traverse_obj(config, ('data', 'attributes', 'chapters', 'elements', ...)):
            start_time = int_or_none(chapter.get('startTime'))
            if not start_time:
                continue
            chapters.append({
                'start_time': start_time,
                'title': chapter.get('title')
            })
```

Reply:
Whoops, guess I'll have to go fix up another extractor too then. Didn't know core does that, thanks for letting me know!